src/graphics/gfx_print.cpp

```diff
@@ -75,25 +93,59 @@ int SuperTextPixLen(int SuperN, const char* SuperChars, int Font)
 
 void SuperPrintRightAlign(int SuperN, const char* SuperChars, int Font, float X, float Y, float r, float g, float b, float a)
 {
+    int dFont = FontManager::fontIdFromSmbxFont(Font);
+    if(NewFontRender && dFont >= 0)
+    {
+        const auto out = std::string(SuperChars).substr(0, SuperN);
```

Comment:
Probably better to make your functions accept sized buffers. There are cases in the config screen where I make views of ~40 strings and I’d rather they not be reallocated every frame.

Reply:
We could also consider upgrading everything to I use string_views but that’s a bit complex for me.

Reply:
Also, string views are not compatible to C++11 which will make game being not buildable on older compilers (currently the minimum compiler version is GCC 4.9, didn't tested on GCC 4.8, so, don't know). But possible to make our own. I want to update these parts too, currently having this as a temporary solution.

Reply:
Okay, then I think let's just use sized buffers. Would you like me to implement those call signatures, or you?

Reply:
Just now I checked this and I reminded: this MESS I made as a lazy workaround because I didn't made the "textSize" that accepts the char* pointer and the size value, I going to definitely update this to avoid having such mess in future.

Reply:
Okay, I completely turned the whole into const char* and size_t pair everywhere, should be better now.